rasa/core/policies/ted_policy.py

```diff
@@ -895,15 +896,37 @@ def _check_data(self) -> None:
 
     def _prepare_layers(self) -> None:
         for name in self.data_signature.keys():
-            self._prepare_sparse_dense_layer_for(name, self.data_signature)
             if name in SEQUENCE_FEATURES_TO_ENCODE:
-                self._prepare_sequence_layers(name)
+                self._tf_layers[
+                    f"{name}_sequence_layer"
+                ] = rasa_layers.RasaSequenceLayer(
+                    name, self.data_signature[name], self.config
+                )
+            elif SENTENCE in self.data_signature[name]:
+                self._tf_layers[f"{name}_input_layer"] = rasa_layers.RasaInputLayer(
+                    name, self.data_signature[name], self.config
+                )
+            else:
+                logger.debug(
+                    f"Creationg no feature combining layers for attribute '{name}' with data signature: {self.data_signature[name]}"
+                )
             self._prepare_encoding_layers(name)
 
         for name in self.label_signature.keys():
-            self._prepare_sparse_dense_layer_for(name, self.label_signature)
             if name in SEQUENCE_FEATURES_TO_ENCODE:
-                self._prepare_sequence_layers(name)
+                self._tf_layers[
+                    f"{name}_sequence_layer"
+                ] = rasa_layers.RasaSequenceLayer(
+                    name, self.label_signature[name], self.config
+                )
+            elif SENTENCE in self.label_signature[name]:
+                self._tf_layers[f"{name}_input_layer"] = rasa_layers.RasaInputLayer(
```

Comment:
why do you use input layer? why not sparse dense directly? there is nothing to concatenate

Reply:
I've now changed this to use the sparse+dense layer directly, thanks!